zerver/views/user_settings.py

```diff
@@ -34,8 +35,13 @@
     EmailChangeStatus
 from confirmation.models import EmailChangeConfirmation
 
+@zulip_login_required
 def confirm_email_change(request, confirmation_key):
     # type: (HttpRequest, str) -> HttpResponse
+    user_profile = request.user
+    if user_profile.realm.email_changes_disabled:
+        raise JsonableError(_("Email change disabled"))
+
```

Comment:
An updated thought on this -- I think only the `@zulip_login_required` should be in the other commit.